### PR TITLE
Api=> getApprovedGatepass - warden

### LINKED ISSUE #79: Warden Approved API
make an API that fetches all the approved gatepasses by the warden who has logged in.

### PR DESCRIPTION
**Pull request by karamarora20**
api created and tested

### CODE REVIEW

File: server-side/node-mssql-crud/src/database/query.js [hunk 1]
```diff
@@ -93,6 +93,7 @@ export const queries = {
 
   acceptGatepass: "UPDATE [gps_db].[gps_db].[gps_gatepassmaster] SET status='Approved' WHERE request_id=@id;",
   rejectGatepass: "UPDATE [gps_db].[gps_db].[gps_gatepassmaster] SET status='Rejected' WHERE request_id=@id;",
+  getApprovedGatepass: "SELECT name, gps_db.gps_gatepass_type.gatepass_name,approved_or_rejected_date, [gps_db].[gps_db].[gps_gatepassmaster].status FROM [gps_db].[gps_db].[gps_gatepassmaster] JOIN gps_db.gps_gatepass_type ON gps_gatepassmaster.gatepass_type = gps_gatepass_type.gatepass_type JOIN gps_db.gps_usersmaster ON gps_gatepassmaster.user_id= gps_db.gps_usersmaster.user_id WHERE send_approval_to=@user_id and ([gps_db].[gps_db].[gps_gatepassmaster].status='Approved' or [gps_db].[gps_db].[gps_gatepassmaster].status='CHECKEDIN');",
```
Comment (palashmunshi) [inline server-side/node-mssql-crud/src/database/query.js]: Use aliasing for better readability. refer getCheckedOutStudents query.

**Review by palashmunshi — CHANGES_REQUESTED** [review]

File: server-side/node-mssql-crud/src/routes/warden.routes.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import { Router } from "express";
-import { 
-    gatepassApproveOrReject, } from "../controllers/warden.controller"
+import { gatepassApproveOrReject, } from "../controllers/warden.controller"
+import { getApprovedGatepass } from "../controllers/warden.controller"
 
 const router = Router();
 
 router.get('/warden/gatepass_approve_or_reject/:id/:flag', gatepassApproveOrReject);
+router.get('/warden/getApprovedGatepass/:user_id', getApprovedGatepass);
```
Comment (palashmunshi) [inline server-side/node-mssql-crud/src/routes/warden.routes.js]: what is the :user_id parameter for??

**Review by palashmunshi — CHANGES_REQUESTED** [review]

File: server-side/node-mssql-crud/src/database/query.js (hunk 1 above)
Comment (palashmunshi) [inline server-side/node-mssql-crud/src/database/query.js]: also what about a case where the student is checked out? will his entry show in this list?

File: server-side/node-mssql-crud/src/routes/warden.routes.js (hunk 2 above)
Comment (karamarora20) [reply server-side/node-mssql-crud/src/routes/warden.routes.js]: for getting the particular warden's user id

File: server-side/node-mssql-crud/src/database/query.js (hunk 1 above)
Comment (karamarora20) [reply server-side/node-mssql-crud/src/database/query.js]: was not found in the examples found but has been added in the latest comment

Comment (karamarora20) [reply server-side/node-mssql-crud/src/database/query.js]: done

**Review by palashmunshi — APPROVED** [review]
nice

**Pull request merged**